src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/ObjectExplorerService.cs

```diff
@@ -0,0 +1,353 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.SqlTools.Extensibility;
+using Microsoft.SqlTools.Hosting;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Connection.Contracts;
+using Microsoft.SqlTools.ServiceLayer.ObjectExplorer.Contracts;
+using Microsoft.SqlTools.ServiceLayer.ObjectExplorer.Nodes;
+using Microsoft.SqlTools.ServiceLayer.ObjectExplorer.SmoModel;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectExplorer
+{
+    /// <summary>
+    /// A Service to support querying server and database information as an Object Explorer tree.
+    /// The APIs used for this are modeled closely on the VSCode TreeExplorerNodeProvider API.
+    /// </summary>
+    public class ObjectExplorerService : HostedService<ObjectExplorerService>, IComposableService
+    {
+        internal const string uriPrefix = "objectexplorer://";
+        
+        // Instance of the connection service, used to get the connection info for a given owner URI
+        private ConnectionService connectionService;
+        private IProtocolEndpoint serviceHost;
+        private Dictionary<string, ObjectExplorerSession> sessionMap;
+        private readonly Lazy<Dictionary<string, HashSet<ChildFactory>>> applicableNodeChildFactories;
+        private IMultiServiceProvider serviceProvider;
+        private static ObjectExplorerService instance = new ObjectExplorerService();
```

Comment:
>instance [](start = 45, length = 8)

With the new model, we don't need an instance. Can we remove this in a subsequent PR? The FormatterService has good examples of this, anything like the child factories can get the service from the service provider so no need for this anymore.